drodrpg/DROD/MapWidget.cpp

```diff
@@ -927,9 +927,14 @@ bool CMapWidget::SelectRoomIfValid(
 			dwRoomX, dwRoomY);
 	if (dwRoomID) //Yes.
 	{
+		MapState storedMapState = MapState::Invisible;
+		if (this->pCurrentGame && this->pCurrentGame->pTotalMapStates)
+			storedMapState = this->pCurrentGame->pTotalMapStates->GetStoredMapStateForRoom(dwRoomID);
 		//Has it been explored?
-		if (this->bEditing || this->ExploredRooms.has(dwRoomID) ||
-				this->NoDetailRooms.has(dwRoomID)) //Yes.
+		if (this->bEditing ||
+			this->ExploredRooms.has(dwRoomID) ||
+			this->NoDetailRooms.has(dwRoomID) ||
+			storedMapState != MapState::Invisible) //Yes.
```

Comment:
PSA: If-clause checks are performed from left to right. As an optimization, we can put the cheaper checks first. Let's put this new check (which might be a single CPU instruction) before the complex data structure scans.

Reply:
Noticed this comment is still outstanding here. Let's address. After that, do you know how to squash the set of commits into a single commit?